presto-main/src/main/java/com/facebook/presto/metadata/Metadata.java

```diff
@@ -73,6 +73,11 @@ public interface Metadata
 
     Optional<SystemTable> getSystemTable(Session session, QualifiedObjectName tableName);
 
+    /**
+     * Returns a table handle for time travel expression
+     */
+    Optional<TableHandle> getTimeTravelHandle(Session session, QualifiedObjectName tableName, Optional<Type> timeTravelType, Optional<Object> timeTravelExpression);
```

Comment:
Once again, I would avoid the term time travel, and perhaps reference versioning.  Perhaps `getHandleVersion`.  Git semantics, not Back To The Future semantics. :)